homeassistant/components/transmission/config_flow.py

```diff
@@ -51,27 +73,17 @@ def async_get_options_flow(config_entry):
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
-
         if user_input is not None:
-
-            for entry in self._async_current_entries():
-                if (
-                    entry.data[CONF_HOST] == user_input[CONF_HOST]
-                    and entry.data[CONF_PORT] == user_input[CONF_PORT]
-                ):
-                    return self.async_abort(reason="already_configured")
-                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
-                    errors[CONF_NAME] = "name_exists"
-                    break
-            try:
-                await get_api(self.hass, user_input)
-
-            except AuthenticationError:
-                errors[CONF_USERNAME] = "invalid_auth"
-                errors[CONF_PASSWORD] = "invalid_auth"
-            except (CannotConnect, UnknownError):
-                errors["base"] = "cannot_connect"
-
+            await self.async_set_unique_id(
+                f"{user_input[CONF_HOST]}-{user_input[CONF_PORT]}"
```

Comment:
host/port cannot be used to define the unique_id

If there is no unique_id available, then the previous method was the correct way to implement it